jwql/jwql_monitors/generate_preview_images.py

```diff
@@ -529,16 +540,23 @@ def get_base_output_name(filename_dict):
 
 @log_fail
 @log_info
-def generate_preview_images():
+def generate_preview_images(overwrite):
     """The main function of the ``generate_preview_image`` module.
-    See module docstring for further details."""
+    See module docstring for further details.
+
+    Parameters
+    ----------
+    overwrite : bool
+        If True, any existing preview images and thumbnails are overwritten
+    """
 
     # Process programs in parallel
     program_list = [os.path.basename(item) for item in glob.glob(os.path.join(SETTINGS['filesystem'], 'public', 'jw*'))]
     program_list.extend([os.path.basename(item) for item in glob.glob(os.path.join(SETTINGS['filesystem'], 'proprietary', 'jw*'))])
     program_list = list(set(program_list))
+    program_list = [(element, overwrite) for element in program_list]
     pool = multiprocessing.Pool(processes=int(SETTINGS['cores']))
-    results = pool.map(process_program, program_list)
+    results = pool.starmap(process_program, program_list)
```

Comment:
Since each tuple will have the same second parameter in the list, you could use the program_list with map (instead of starmap), and just send overwrite as the 'y' parameter.  This wouldn't change the functionality, just simplifies the code.  This is fine as is, just giving feedback.

Reply:
I'm not sure I understand. map() only accepts one parameter, right? So in that case I'd have to also make program_list a tuple, and I'd have to change the parameters of process_program() such that it accepts a single parameter that is then broken out into program id and overwrite status, right? Or am I misinterpreting the way that map() works?

Reply:
I withdraw my comment, Map can take multiple parameters, but they both need to be iterable anyway, this way makes more sense.